### PR TITLE
Add function to allow nameplate access to the collected bg information

### PR DESCRIPTION
**Pull request by xeropresence**
Add function to allow nameplate access to the collected bg information. This will allow a more accurate healer indicator since we can use the same data used for the spec icon.

### CODE REVIEW

File: Plater.lua [hunk 1]
```diff
@@ -8259,6 +8259,16 @@ end
 		end
 	end
 	
+	function Plater.GetBgPlayerRoleCache(unit)
+		local name = GetUnitName(unit, true)
+		if not BG_PLAYER_CACHE[name] then
+			Plater.UpdateBgPlayerRoleCache()
+		end
+		
+		local cache = BG_PLAYER_CACHE[name]
```
Comment (cont1nuity) [inline Plater.lua:8268]: Simple `return BG_PLAYER_CACHE[name]` suffices, no need for another local in this case.

File: Plater.lua [hunk 2]
```diff
@@ -11301,4 +11311,4 @@ function Plater.OpenColorFrame()
 	a:SetWidth (totalWidth)
 end
 
---functiona enda
```
Comment (cont1nuity) [inline Plater.lua:11304]: Why the change to EOF (i assume)? Please revert.

File: Plater.lua [hunk 3]
```diff
@@ -8259,6 +8259,16 @@ end
 		end
 	end
 	
+	function Plater.GetBgPlayerRoleCache(unit)
```
Comment (cont1nuity) [inline Plater.lua:8262]: I think a more appropriate name would be `GetPlayerBGInfo` or `GetUnitBGInfo`

**Comment by cont1nuity** [comment]
Please update the PR with a more fitting name.

File: Plater.lua (hunk 2 above)
Comment (xeropresence) [reply Plater.lua:11304]: Editing via the website force adds eol

**Comment by xeropresence** [comment]
I've updated the commit to change the name of the function and removed the eof eol

**Comment by cont1nuity** [comment]
Thank you for your contribution!
